dace/runtime/include/dace/perf/reporting.h

```diff
@@ -9,73 +9,185 @@
 #include <sstream>
 #include <vector>
 
+#ifdef _WIN32
+#include <process.h>
+#endif
+
+#ifdef __unix__
+#include <unistd.h>
+#endif
+
+#define DACE_REPORT_BUFFER_SIZE 2048
+
 namespace dace {
 namespace perf {
 
+    struct TraceEvent {
+        char ph;
+        std::string name;
+        std::string cat;
```

Comment:
wouldn't dynamically-sized strings be slow?

Reply:
Uses fixed char array buffering now, seems to indeed be minutely faster.